aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, String catalogId, String warehousePath) {
+    this(conf, GlueClient.create(), catalogId, warehousePath);
+  }
+
+  /**
+   * constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, GlueClient glue, String catalogId, String warehousePath) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.warehousePath = warehousePath;
+    validateWarehousePath();
+  }
+
+  private void validateWarehousePath() {
+    Preconditions.checkArgument(warehousePath != null && warehousePath.length() > 0,
+        "Glue warehouse path must be set");
+    Preconditions.checkArgument(warehousePath.charAt(warehousePath.length() - 1) != '/',
+        "Glue warehouse path %s must not end with /", warehousePath);
+  }
+
+  @Override
+  public void close() throws IOException {
+    glue.close();
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    String dbName = IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace());
+    String tableName = tableIdentifier.name();
+    return new GlueTableOperations(conf, glue, catalogId, dbName, tableName);
+  }
+
+  /**
+   * Use the same format as Hive for default warehouse location
+   * @param tableIdentifier table id
+   * @return default warehouse path
+   */
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    // check if value is set in database
+    GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+        .name(IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()))
+        .build());
+    String dbLocationUri = response.database().locationUri();
+    if (dbLocationUri != null) {
+      return String.format("%s/%s", dbLocationUri, tableIdentifier.name());
+    }
+
+    return String.format(
+        "%s/%s.db/%s",
+        warehousePath,
+        IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()),
+        tableIdentifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    namespaceExists(namespace);
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<TableIdentifier> results = new ArrayList<>();
+    while (true) {
+      GetTablesResponse response = glue.getTables(GetTablesRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.toDatabaseName(namespace))
+          .nextToken(nextToken)
+          .build());
```

Comment:
You might need a `try ... catch` block here as the call to `getTables` can throw iirc.

Reply:
I did not catch any exception because glue client only throw RuntimeException, and it seems to be okay in other catalog implementations.

Reply:
It depends on the exception. If the namespace is missing, then we have an exception for that so users can handle it. We would want to translate the AWS exception to the Iceberg one. But for connection issues, it's okay to throw the original exception.